aspnetcore/blazor/fundamentals/routing.md

```diff
@@ -1902,6 +1920,110 @@ In the following `App` component example:
 > [!NOTE]
 > Not throwing if the cancellation token in <xref:Microsoft.AspNetCore.Components.Routing.NavigationContext> is canceled can result in unintended behavior, such as rendering a component from a previous navigation.
 
+## Handle location changes in a component
+
+`RegisterLocationChangingHandler` registers a handler to process incoming navigation events. The handler's context provided by `LocationChangingContext` includes the following properties:
+
+* `TargetLocation`: Gets the target location.
+* `HistoryEntryState`: Gets the state associated with the target history entry.
+* `IsNavigationIntercepted`: Gets whether the navigation was intercepted from a link.
+* `CancellationToken`: Gets a <xref:System.Threading.CancellationToken> to determine if the navigation was canceled, for example, to determine if the user triggered a different navigation.
+* `PreventNavigation`: Called to prevent the navigation from continuing.
+
+A component can register multiple location changing handlers, and calling <xref:Microsoft.AspNetCore.Components.NavigationManager.NavigateTo%2A> invokes all of the registered handlers. Registered handlers are automatically disposed by the Navigation Manager when the component is disposed.
```

Comment:
> A component can register multiple location changing handlers, and calling `Microsoft.AspNetCore.Components.NavigationManager.NavigateTo` invokes all of the registered handlers.

While this is true, navigations will invoke _all_ location changing handlers registered across the entire app (i.e. handlers can be registered across multiple components, and any internal navigation will execute them all in parallel). Also, in addition to `NavigateTo()` invoking the handlers, clicking on internal links (those that point to URLs under the base href) and navigating using the forward/back buttons will also invoke the handlers.

It's also probably worth pointing out that these handlers will only get run for internal navigations within the app. If you click on a link that takes you to a different site (or change the address bar to a different site manually, etc.), then location changing handlers will not be executed. However, the `<NavigationLock />` component does have a `ConfirmExternalNavigation` parameter that catches these cases and displays the built-in browser "Are you sure you want to leave this page?" dialog.